### PR TITLE
[new-feature] Added approval module

### PR DESCRIPTION
**Pull request by SHRE-YANSH**
This module allows whitelisting users from flood, warns. locks on a per group basis.

### CODE REVIEW

File: SaitamaRobot/modules/approve.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+from telegram.ext import CallbackContext, run_async, Update, CallbackQueryHandler
+import SaitamaRobot.modules.sql.approve_sql as sql
+from SaitamaRobot.modules.helper_funcs.chat_status import (
+    bot_admin, user_admin)
+from telegram import ParseMode, InlineKeyboardMarkup, InlineKeyboardButton
+
+
+@user_admin
+@run_async
+def approve(update, context):
```
Comment (DragSama) [inline SaitamaRobot/modules/approve.py:17]: This should be `@loggable`  to make sure log channel get the info, Also add type hints if possible

File: SaitamaRobot/modules/approve.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        message.reply_text(
+            f"[{member.user['first_name']}](tg://user?id={member.user['id']}) is already approved in {chat_title}", parse_mode=ParseMode.MARKDOWN)
+        return
+    sql.approve(message.chat_id, user_id)
+    message.reply_text(f"[{member.user['first_name']}](tg://user?id={member.user['id']}) has been approved in {chat_title}! They will now be ignored by automated admin actions like locks, blocklists, and antiflood.", parse_mode=ParseMode.MARKDOWN)
+
+
+@user_admin
+@run_async
+def disapprove(update, context):
```
Comment (DragSama) [inline SaitamaRobot/modules/approve.py:62]: This should also be `@loggable`

File: SaitamaRobot/modules/approve.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            approved_users = sql.list_approved(chat.id)
+            for i in approved_users:
+                users.append(int(i.user_id))
+            for user_id in users:
+                sql.disapprove(chat.id, user_id)
+
+        if member.status == "administrator":
+            query.answer("Only owner of the chat can do this.")
+
+        if member.status == "member":
```
Comment (DragSama) [inline SaitamaRobot/modules/approve.py:188]: I don't think there is need for two if, You can just say "Only owner of the chat can do this." to both admin and member

File: SaitamaRobot/modules/approve.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+dispatcher.add_handler(APPROVE)
+dispatcher.add_handler(DISAPPROVE)
+dispatcher.add_handler(APPROVED)
+dispatcher.add_handler(APPROVAL)
+dispatcher.add_handler(UNAPPROVEALL)
+dispatcher.add_handler(UNAPPROVEALL_BTN)
+
+__mod_name__ = "Approvals"
+__command_list__ = ["approve", "unapprove", "approved", "approval"]
+__handlers__ = [APPROVE, DISAPPROVE, APPROVED, APPROVAL]
```
Comment (DragSama) [inline SaitamaRobot/modules/approve.py:230]: Add a newline at EOF to comply with pep8

**Review by DragSama — CHANGES_REQUESTED** [review]

File: SaitamaRobot/modules/approve.py (hunk 3 above)
Comment (SHRE-YANSH) [reply SaitamaRobot/modules/approve.py:188]: Just tried to do like Rose.

Comment (DragSama) [reply SaitamaRobot/modules/approve.py:188]: I don't think that changes the fact that It's Invalid info. You are telling to a member that "You need to be a admin" and telling to a admin that "You need to be owner".

**Review by DragSama — APPROVED** [review]

**Pull request merged**